packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1078,6 +1078,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
         this.summaryCollection = new SummaryCollection(this.deltaManager, this.logger);
 
+        this.dirtyContainer = this.context.attachState !== AttachState.Attached;
+        this.context.updateDirtyContainerState(this.dirtyContainer);
```

Comment:
@wes-carlson , can you please glance? I believe we should set container dirty if this.getPendingLocalState() !== undefined, but I do not see any usage in the code of getPendingLocalState(). Still might be worth bullet-proofing code for future.

Reply:
Only pending ops that can come in through runtime constructor are via `context.pendingLocalState` that gets subsequently passed to `PendingStateManager`. However, PendingStateManager does not consider that set when asked about `hasPendingMessages`. I'll dig in further in the morning.

Reply:
@wes-carlson I'd appreciate your take here. It seems that we can create runtime either in detached state (in which case we don't have any pending ops) or attached state where `context.pendingLocalState` will list out stashed ops that need to be moved to pending state. If we want to be very accurate on `dirty` state we probably need to consider both `context.attachState` and  `context.pendingLocalState`.  

However, within runtime, both logic flows (detached and attached) funnel through replaying pending states when `join` triggers the connection state change, at which point dirty flag is correctly cleared based on the fact that all pending ops just have been sent through. So alternatively, we could default runtime to dirty state (in the constructor), and let the connection event flow clear the flag.

Maybe we could chat briefly on use cases I missed here (ex. implications of read-only to r/w transitions).

Reply:
I think it makes sense to look at context.pendingLocalState. I don't think we'd want to default to dirty since if there are no stashed ops it will be incorrect until `join` happens.

Reply:
Now including `context.pendingLocalState` in addition to attached state. In conclusion, the runtime construction logic:
* Detached context -> Dirty 
* Attached context + pending ops -> Dirty 
* Attached context + no pending ops -> Not Dirty 

`local-server-tests\src\test\documentDirty.spec.ts` && `test-end-to-end-tests\src\test\containerDirtyFlag.spec.ts` should validate all the use cases, inclusive of the ones we discussed here.

Reply:
Can we add this as a comment where we initialize the dirty state. It will be super helpful and clear :)